### PR TITLE
Prohibit usage of vni zero

### PR DESCRIPTION
**Pull request by jetzorkj**

### CODE REVIEW

File: pkg/reconciler/layer3.go [hunk 1]
```diff
@@ -165,6 +165,12 @@ func (r *reconcile) createVrfConfigMap(l3vnis []networkv1alpha1.VRFRouteConfigur
 			continue
 		}
 
+		if vni == 0 {
```
Comment (Cellebyte) [inline pkg/reconciler/layer3.go:168]: Should we not do a boundary check here? 

`vni <= 0 and vni >= max_vni`

**Comment by MaxRink** [comment]
I mean the apiserver can also do that enforcement with a block directly

https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/

Marcel Fest ***@***.***> schrieb am Fr., 15. März 2024, 21:12:

**Comment by Cellebyte** [comment]
@MaxRink only for the CRD ^^ the ConfigMap needs validation as well :)

**Comment by MaxRink** [comment]
Make the ConfigMap an CRD :P


Am Sa., 16. März 2024 um 11:25 Uhr schrieb Marcel Fest <
***@***.***>:

File: config/crd/bases/network.schiff.telekom.de_layer2networkconfigurations.yaml [hunk 2]
```diff
@@ -126,6 +126,7 @@ spec:
                 x-kubernetes-map-type: atomic
               vni:
                 description: VXLAN VNI Id for the layer 2 network
+                minimum: 1
```
Comment (Cellebyte) [inline config/crd/bases/network.schiff.telekom.de_layer2networkconfigurations.yaml:129]: we should add the `maximum: 16777215` as well to also validate the upper boundary.

File: pkg/reconciler/layer3.go [hunk 3]
```diff
@@ -165,6 +165,12 @@ func (r *reconcile) createVrfConfigMap(l3vnis []networkv1alpha1.VRFRouteConfigur
 			continue
 		}
 
+		if vni == 0 && vni > 16777215 {
+			err := fmt.Errorf("VNI can not be set to 0")
```
Comment (Cellebyte) [inline pkg/reconciler/layer3.go:169]: please also add the upper boundary in the error message

File: api/v1alpha1/layer2networkconfiguration_types.go [hunk 4]
```diff
@@ -37,6 +37,7 @@ type Layer2NetworkConfigurationSpec struct {
 	MTU int `json:"mtu"`
 
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
```
Comment (Cellebyte) [inline api/v1alpha1/layer2networkconfiguration_types.go:40]: you need to regenerate the types again

**Pull request merged**